grpc/server.go

```diff
@@ -48,20 +46,43 @@ func NewServer(c *cmd.GRPCServerConfig, tls *tls.Config, serverMetrics *grpc_pro
 	if maxConcurrentStreams == 0 {
 		maxConcurrentStreams = 250
 	}
-	si := &serverInterceptor{serverMetrics}
+	si := newServerInterceptor(metrics, clk)
 	return grpc.NewServer(
 		grpc.Creds(creds),
 		grpc.UnaryInterceptor(si.intercept),
 		grpc.MaxConcurrentStreams(uint32(maxConcurrentStreams)),
 	), l, nil
 }
 
-// NewServerMetrics constructs a *grpc_prometheus.ServerMetrics, registered with
-// the given registry, with timing histogram enabled. It must be called a
+// serverMetrics is a struct type used to return a few registered metrics from
+// `NewServerMetrics`
+type serverMetrics struct {
+	GRPCMetrics *grpc_prometheus.ServerMetrics
```

Comment:
The two fields of this struct should be unexported, since they are not used outside this package.